agent/config_endpoint_test.go

```diff
@@ -95,6 +108,37 @@ func TestConfig_Get(t *testing.T) {
 		_, err := a.srv.Config(resp, req)
 		require.Error(t, errors.New("Must provide either a kind or both kind and name"), err)
 	})
+	t.Run("get the single mesh config", func(t *testing.T) {
+		req, _ := http.NewRequest("GET", "/v1/config/mesh/mesh", nil)
+		resp := httptest.NewRecorder()
+		obj, err := a.srv.Config(resp, req)
+		require.NoError(t, err)
+
+		ce, ok := obj.(*structs.MeshConfigEntry)
+		require.True(t, ok, "wrong type %T", obj)
+		// Set indexes to expected values for assertions
+		ce.CreateIndex = 12
+		ce.ModifyIndex = 13
+
+		out, err := a.srv.marshalJSON(req, obj)
+		require.NoError(t, err)
```

Comment:
It is unfortunate that our API tests don't test against a marshalled response. That seems like a potentially large place for bugs to hide. Here I am calling the underlying function that our HTTP framework calls. It's probably an adequate simulation for now, but seems like it could break in the future.

Also not being able to test for the response code or response headers is really unfortunate.

Reply:
Alternatively you could add a case to `TestAPI_ConfigEntries` in the api module. Wouldn't that verify that the kind was serialized?